### ISSUE TITLE
nrzap log in context ignores additional fields

### ISSUE THREAD

**Issue by NicholasGWK** [newrelic/go-agent#768]
## Description

I'm unsure if this is intended behaviour or not, but while debugging I noticed the nrzap logs in context integration does NOT do anything with zap fields...my initial expectation was that these would be added as custom attributes to the log entries.

The code here shows that the fields are just not used: https://github.com/newrelic/go-agent/blob/046b4fc116adac2b4dec4d691b61a25b80488fc2/v3/integrations/logcontext-v2/nrzap/nrzap.go#L28

Please let me know if this intended or if there's another way to add these fields as attributes!

## Steps to Reproduce

- Setup nrzap integration
- Create a transaction logger similar to the example: https://github.com/newrelic/go-agent/blob/046b4fc116adac2b4dec4d691b61a25b80488fc2/v3/integrations/logcontext-v2/nrzap/example/main.go#L35
- Log a message with additional zap fields, i.e. instead of:

```txnLogger.Error("Some error!")```

do

``` txnLogger.Error("Some error!", zap.Int("http.statusCode", http.StatusInternalServerError)```

## Expected Behavior

The log entry in New Relic would have a custom attributes, "http.statusCode" with the passed value.

I didn't include env info because I'm not sure it matters based on the code I linked to, but please let me know if any additional info is needed!

Thanks so much,

Nick

**Comment by stefanoschrs** [comment]
Do we have any updates on this?

Adding some fields to both logs in the example doesn't work for me either as intended:
```
	backgroundLogger.Info("this is a background log message", zap.String("foo", "bar"))
...
	txnLogger.Info("this is a transaction log message", zap.String("foo", "bar"))
```

Gives this in the console:
```
{"level":"info","ts":1695784536.173117,"msg":"this is a background log message","foo":"bar"}
{"level":"info","ts":1695784536.173432,"msg":"this is a transaction log message","foo":"bar"}
```

but in New Relic:
```
{
  "entity.guid": "xxx",
  "entity.guids": "xxx",
  "entity.name": "xxx",
  "hostname": "xxx",
  "level": "info",
  "message": "this is a background log message",
  "newrelic.source": "xxx",
  "timestamp": xxx
}

{
  "entity.guid": "xxx",
  "entity.guids": "xxx",
  "entity.name": "xxx",
  "hostname": "xxx",
  "level": "info",
  "message": "this is a transaction log message",
  "newrelic.source": "xxx",
  "span.id": "xxx",
  "timestamp": xxx,
  "trace.id": "xxx"
}
```

**Comment by iamemilio** [comment]
Yeah, we are aware of this. There is a roadmap item to add support for logs in context fields in the future, and a possible re-write of the zap libraries that will pair with it.

**Comment by stefanoschrs** [comment]
Thanks for the reply, is there a very rough estimation on when will this happen? Just to know if it's worth waiting or switching to the REST agent way.

**Comment by adomaskizogian** [comment]
bumping this
just run into the same issue with slog.

it is sad that such core functionality is missing and forces us to do parsing of logs on ui

**Comment by iamemilio** [comment]
Hi all, thank you for your advocacy and your patience. I hear your concerns and have gotten the go ahead to prioritize this **ASAP**. Supporting attribute collection for all v2 logs in context packages is my top priority. I will communicate upcoming changes here and in #833.

**Comment by iamemilio** [comment]
I am working on adding support for attribute collection to the agent now. Once I have that code settled, and tested, I will prioritize Zap and then Slog for attribute collection support first. Zerolog's opacity will make attribute collection more difficult.

**Comment by iamemilio** [comment]
Entering the review phase for attribute collection support in the agent: https://github.com/newrelic/go-agent/pull/900. Building this into wrappers for logging frameworks and additional optimizations are on their way. This should make the next release. Feel free to leave feedback on this PR until then.

Notes:
- capture of structs, maps, slices, and arrays are included and will be encoded as a JSON string. The backend will parse and decode these strings. If the JSON string exceeds 256 bytes, it will be truncated and will show up as just a truncated string.
- passing complex objects is probably faster if they are encoded to a string in the logging framework since those tools have much lower allocation costs for constructing JSON strings.

**Comment by mirackara** [comment]
#902 is the WIP PR specifically for zap attribute collection. We are currently in the testing phase for attribute collection for zap. Along with #900, this should also be in the next release. Feel free to leave feedback/suggestions on this PR as well!

**Comment by iamemilio** [comment]
This feature has been released for Zap in the [3.33.0](https://github.com/newrelic/go-agent/releases/tag/v3.33.0) version of the agent! https://github.com/newrelic/go-agent/pull/908
